ADAP incorporated into MZmine 2

du-lab opened the pull request:
Dear MZmine 2 development team,

My Du-lab team have incorporated two new modules we developed into MZmine 2. One module carries out chromatogram building and the other module carries out chromatogram deconvolution. These two modules can be used for both LC-MS and GC-MS data processing.

In addition to these additions to MZmine 2, we have also made some changes to MZmine 2 itself. Please see page 36 of the attached ADAP User Manual for the list of major changes.

There are also some minor changes we made to MZmine 2 that are not listed on this page.

In addition to the list of major changes, this pdf attachment also includes a step-by-step procedure about how to use the two new modules in the context of MZmine 2. 

Please ignore Chapter 4 "ADAP-GC" for now.

[ADAP_user_manual.pdf](https://github.com/mzmine/mzmine2/files/828729/ADAP_user_manual.pdf)

Thank you very much for reviewing the additions and changes. Please let me know if you have any questions.

Best regards,
Xiuxia Du
xiuxia.du@uncc.edu
704-687-7307

tomas-pluskal commented:
Hi Xiuxia,
It seems the pull request has some conflicts, so it cannot be merged in directly. The first 3 commits look ok, but the following 4 commits have some conflicts. Could you try to prepare the pull request again by resolving the conflicts?
Also, I am not sure about the commit "Removed Alignment, Peakpicking, Decomposition, and DBSearch modules from MZmine interface.". What is the purpose?
Tomas

tomas-pluskal commented:
Perhaps the easiest way to proceed would be to start with a fresh checkout of the MZmine master branch, and add your new files in several stages. In the first stage, we could process all the additions (new modules) that do not require changes to existing code (e.g., adding dependencies to pom.xml, adding new MZmine modules, adding new functions etc). In the second stage, we can process the changes to existing code - these have to be done carefully. What do you think?

tomas-pluskal commented:
Wait... I found a way how to merge your pull request in.

tomas-pluskal commented:
OK, so now I can merge the pull request. But I have a few questions:
- What is the reason so many jar files were added to the lib/ folder? We normally use maven for dependencies. 
- Why is adap.jar included as a dependency, if the source code is also included in the source tree? What is actually in the adap.jar?
- Why are some of the existing modules commented out in MZmineModulesList.java?

tomas-pluskal commented on pom.xml:
```diff
@@ -396,7 +396,16 @@
+        </dependency>
+        <dependency>
+            <groupId>dulab</groupId>
+            <artifactId>adap</artifactId>
```
Could you please describe what is the purpose of this dependency? Shouldn't the code be included directly in MZmine instead?

tomas-pluskal commented on src/main/java/net/sf/mzmine/datamodel/impl/SimpleDataPoint.java:
```diff
@@ -59,6 +59,10 @@ public double getMZ() {
 	return mz;
     }
 
+    public void setIntensity(final double intensity) {
```
The setIntensity() method should not be here. Data points are immutable by design (as described in the class javadoc), same as other basic Java classes (String, Double, etc.). Instead of changing the intensity, you should always create a new data point instance.

tomas-pluskal commented on src/main/java/net/sf/mzmine/modules/visualization/twod/help/help.html~HEAD:
```diff
@@ -0,0 +1,105 @@
+<html>
```
I suppose this file is a leftover from some merging?

tomas-pluskal commented on src/main/java/net/sf/mzmine/parameters/parametertypes/.LCKParameterSetComponents.java~:
```diff
@@ -0,0 +1 @@
+/Users/aleksandrsmirnov/Projects/peak_picking/owen-mod-mzmine2-master/src/main/java/net/sf/mzmine/parameters/parametertypes/ParameterSetComponents.java
```
Temporary file, please remove..

tomas-pluskal commented:
Hi Xiuxia,

I requested some changes in the pull request your lab submitted (
https://github.com/mzmine/mzmine2/pull/278). Please take a look at the
comments and let me know.

Best,

Tomas



On Thu, Mar 9, 2017 at 4:20 AM, du-lab <notifications@github.com> wrote:

> Dear MZmine 2 development team,
>
> My Du-lab team have incorporated two new modules we developed into MZmine
> 2. One module carries out chromatogram building and the other module
> carries out chromatogram deconvolution. These two modules can be used for
> both LC-MS and GC-MS data processing.
>
> In addition to these additions to MZmine 2, we have also made some changes
> to MZmine 2 itself. Please see page 36 of the attached ADAP User Manual for
> the list of major changes.
>
> There are also some minor changes we made to MZmine 2 that are not listed
> on this page.
>
> In addition to the list of major changes, this pdf attachment also
> includes a step-by-step procedure about how to use the two new modules in
> the context of MZmine 2.
>
> Please ignore Chapter 4 "ADAP-GC" for now.
>
> ADAP_user_manual.pdf
> <https://github.com/mzmine/mzmine2/files/828729/ADAP_user_manual.pdf>
>
> Thank you very much for reviewing the additions and changes. Please let me
> know if you have any questions.
>
> Best regards,
> Xiuxia Du
> xiuxia.du@uncc.edu
> 704-687-7307
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/mzmine/mzmine2/pull/278
> Commit Summary
>
>    - temporarily need a repository for adap-in-mzmine src
>    - updating files
>    - modified pom file to make the ADAP package work
>    - Fixed a bug with updates in Decomposition preview
>    - Removed Alignment, Peakpicking, Decomposition, and DBSearch modules
>    from MZmine interface. The corresponding modules are still part of ZMmine
>    though.
>    - Changed the extention for MSP Export from .txt to .msp
>    - Added the most recent ADAP .jar file to MZmine
>
> File Changes
>
>    - *M* pom.xml <https://github.com/mzmine/mzmine2/pull/278/files#diff-0>
>    (11)
>    - *M* src/main/java/net/sf/mzmine/datamodel/DataPoint.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-1> (1)
>    - *M* src/main/java/net/sf/mzmine/datamodel/Feature.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-2> (7)
>    - *M* src/main/java/net/sf/mzmine/datamodel/PeakIdentity.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-3> (1)
>    - *A* src/main/java/net/sf/mzmine/datamodel/PeakInformation.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-4> (46)
>    - *M* src/main/java/net/sf/mzmine/datamodel/PeakListRow.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-5> (22)
>    - *M* src/main/java/net/sf/mzmine/datamodel/RawDataFile.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-6> (4)
>    - *A* src/main/java/net/sf/mzmine/datamodel/impl/.
>    SimplePeakListRow.java.swp
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-7> (0)
>    - *M* src/main/java/net/sf/mzmine/datamodel/impl/SimpleDataPoint.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-8> (4)
>    - *M* src/main/java/net/sf/mzmine/datamodel/impl/SimpleFeature.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-9> (17)
>    - *A* src/main/java/net/sf/mzmine/datamodel/impl/
>    SimplePeakInformation.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-10> (85)
>    - *M* src/main/java/net/sf/mzmine/datamodel/impl/SimplePeakList.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-11> (1)
>    - *M* src/main/java/net/sf/mzmine/datamodel/impl/SimplePeakListRow.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-12> (19)
>    - *M* src/main/java/net/sf/mzmine/main/MZmineModulesList.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-13> (32)
>    - *A* src/main/java/net/sf/mzmine/modules/masslistmethods/.DS_Store
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-14> (0)
>    - *A* src/main/java/net/sf/mzmine/modules/masslistmethods/
>    ADAPchromatogrambuilder/.LCKChromatogramBuilderParameters.java~
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-15> (1)
>    - *A* src/main/java/net/sf/mzmine/modules/masslistmethods/
>    ADAPchromatogrambuilder/ADAPChromatogramBuilderModule.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-16> (79)
>    - *A* src/main/java/net/sf/mzmine/modules/masslistmethods/
>    ADAPchromatogrambuilder/Chromatogram.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-17> (550)
>    - *A* src/main/java/net/sf/mzmine/modules/masslistmethods/
>    ADAPchromatogrambuilder/ChromatogramBuilderParameters.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-18> (78)
>    - *A* src/main/java/net/sf/mzmine/modules/masslistmethods/
>    ADAPchromatogrambuilder/ChromatogramBuilderTask.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-19> (407)
>    - *A* src/main/java/net/sf/mzmine/modules/masslistmethods/
>    ADAPchromatogrambuilder/ExpandedDataPoint.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-20> (66)
>    - *A* src/main/java/net/sf/mzmine/modules/masslistmethods/
>    ADAPchromatogrambuilder/HighestDataPointConnector.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-21> (183)
>    - *M* src/main/java/net/sf/mzmine/modules/masslistmethods/
>    chromatogrambuilder/Chromatogram.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-22> (15)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/
>    alignment/adap3/ADAP3AlignerModule.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-23> (56)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/
>    alignment/adap3/ADAP3AlignerOptimizationParameters.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-24> (54)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/
>    alignment/adap3/ADAP3AlignerParameters.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-25> (77)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/
>    alignment/adap3/ADAP3AlignerTask.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-26> (546)
>    - *M* src/main/java/net/sf/mzmine/modules/peaklistmethods/
>    alignment/ransac/RansacAlignerTask.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-27> (25)
>    - *M* src/main/java/net/sf/mzmine/modules/peaklistmethods/filtering/
>    peakcomparisonrowfilter/PeakComparisonRowFilterTask.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-28> (2)
>    - *M* src/main/java/net/sf/mzmine/modules/peaklistmethods/
>    filtering/rowsfilter/RowsFilterTask.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-29> (6)
>    - *M* src/main/java/net/sf/mzmine/modules/peaklistmethods/
>    gapfilling/samerange/SameRangePeak.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-30> (17)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/
>    identification/adap3GCMSsearch/ADAP3GCMSSearchModule.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-31> (149)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/
>    identification/adap3GCMSsearch/ADAP3GCMSSearchParameters.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-32> (46)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/
>    identification/adap3GCMSsearch/ADAP3GCMSSearchTask.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-33> (130)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/
>    identification/adap3LCMSsearch/ADAP3LCMSSearchModule.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-34> (139)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/
>    identification/adap3LCMSsearch/ADAP3LCMSSearchParameters.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-35> (52)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/
>    identification/adap3LCMSsearch/ADAP3LCMSSearchTask.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-36> (132)
>    - *M* src/main/java/net/sf/mzmine/modules/peaklistmethods/
>    identification/camera/CameraSearchModule.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-37> (2)
>    - *M* src/main/java/net/sf/mzmine/modules/peaklistmethods/
>    identification/camera/CameraSearchParameters.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-38> (45)
>    - *M* src/main/java/net/sf/mzmine/modules/peaklistmethods/
>    identification/camera/CameraSearchTask.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-39> (440)
>    - *M* src/main/java/net/sf/mzmine/modules/peaklistmethods/
>    identification/onlinedbsearch/PeakListIdentificationTask.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-40> (4)
>    - *M* src/main/java/net/sf/mzmine/modules/peaklistmethods/io/
>    csvexport/CSVExportParameters.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-41> (6)
>    - *M* src/main/java/net/sf/mzmine/modules/peaklistmethods/io/
>    csvexport/CSVExportTask.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-42> (90)
>    - *M* src/main/java/net/sf/mzmine/modules/peaklistmethods/io/csvexport/
>    ExportRowDataFileElement.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-43> (4)
>    - *M* src/main/java/net/sf/mzmine/modules/peaklistmethods/io/
>    metaboanalystexport/MetaboAnalystExportTask.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-44> (3)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/io/
>    mgfexport/MGFExportModule.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-45> (70)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/io/
>    mgfexport/MGFExportParameters.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-46> (63)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/io/
>    mgfexport/MGFExportTask.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-47> (224)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/io/
>    mspexport/MSPExportModule.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-48> (70)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/io/
>    mspexport/MSPExportParameters.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-49> (63)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/io/
>    mspexport/MSPExportTask.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-50> (229)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/peakpicking/
>    adap3decomposition/ADAP3DecompositionModule.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-51> (76)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/peakpicking/
>    adap3decomposition/ADAP3DecompositionParameters.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-52> (62)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/peakpicking/
>    adap3decomposition/ADAP3DecompositionTask.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-53> (650)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/peakpicking/
>    adap3decomposition/ADAP3EICDecompositionParameters.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-54> (78)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/peakpicking/
>    adap3decomposition/ADAP3OptimizationParameters.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-55> (65)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/peakpicking/
>    adap3decomposition/ADAP3TICWindowDetectionParameters.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-56> (72)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/peakpicking/
>    adap3decompositionV1_5/ADAP3DecompositionV1_5Module.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-57> (76)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/peakpicking/
>    adap3decompositionV1_5/ADAP3DecompositionV1_5Parameters.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-58> (119)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/peakpicking/
>    adap3decompositionV1_5/ADAP3DecompositionV1_5SetupDialog.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-59> (571)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/peakpicking/
>    adap3decompositionV1_5/ADAP3DecompositionV1_5Task.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-60> (485)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/peakpicking/
>    adap3decompositionV1_5/EICPlot.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-61> (170)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/peakpicking/
>    adap3decompositionV1_5/SimpleScatterPlot.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-62> (116)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/peakpicking/
>    adap3decompositionV2/ADAP3DecompositionV2Module.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-63> (76)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/peakpicking/
>    adap3decompositionV2/ADAP3DecompositionV2Parameters.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-64> (85)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/peakpicking/
>    adap3decompositionV2/ADAP3DecompositionV2Task.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-65> (517)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/peakpicking/
>    adap3peakdetection/ADAP3PeakDetectionModule.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-66> (74)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/peakpicking/
>    adap3peakdetection/ADAP3PeakDetectionParameters.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-67> (85)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/peakpicking/
>    adap3peakdetection/ADAP3PeakDetectionTask.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-68> (448)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/
>    peakpicking/deconvolution/ADAPpeakpicking/ADAPDetector.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-69> (218)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/
>    peakpicking/deconvolution/ADAPpeakpicking/ADAPDetectorParameters.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-70> (136)
>    - *M* src/main/java/net/sf/mzmine/modules/peaklistmethods/
>    peakpicking/deconvolution/DeconvolutionParameters.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-71> (4)
>    - *M* src/main/java/net/sf/mzmine/modules/peaklistmethods/
>    peakpicking/deconvolution/DeconvolutionTask.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-72> (1)
>    - *M* src/main/java/net/sf/mzmine/modules/peaklistmethods/
>    peakpicking/deconvolution/PeakResolverSetupDialog.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-73> (7)
>    - *M* src/main/java/net/sf/mzmine/modules/peaklistmethods/
>    peakpicking/deconvolution/ResolvedPeak.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-74> (29)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/
>    peakpicking/deconvolution/adapnewpeakpickingGC/
>    ADAP3NewGCPeakDetection.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-75> (233)
>    - *A* src/main/java/net/sf/mzmine/modules/peaklistmethods/
>    peakpicking/deconvolution/adapnewpeakpickingGC/
>    ADAP3NewGCPeakDetectionParameters.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-76> (154)
>    - *M* src/main/java/net/sf/mzmine/modules/peaklistmethods/
>    peakpicking/peakextender/ExtendedPeak.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-77> (14)
>    - *M* src/main/java/net/sf/mzmine/modules/peaklistmethods/
>    peakpicking/shapemodeler/peakmodels/EMGPeakModel.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-78> (15)
>    - *M* src/main/java/net/sf/mzmine/modules/peaklistmethods/
>    peakpicking/shapemodeler/peakmodels/GaussianPeakModel.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-79> (15)
>    - *M* src/main/java/net/sf/mzmine/modules/peaklistmethods/
>    peakpicking/shapemodeler/peakmodels/TrianglePeakModel.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-80> (14)
>    - *M* src/main/java/net/sf/mzmine/modules/projectmethods/
>    projectload/version_2_5/PeakListElementName_2_5.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-81> (6)
>    - *M* src/main/java/net/sf/mzmine/modules/projectmethods/
>    projectload/version_2_5/PeakListOpenHandler_2_5.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-82> (39)
>    - *M* src/main/java/net/sf/mzmine/modules/projectmethods/projectsave/
>    PeakListElementName.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-83> (10)
>    - *M* src/main/java/net/sf/mzmine/modules/projectmethods/projectsave/
>    PeakListSaveHandler.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-84> (41)
>    - *A* src/main/java/net/sf/mzmine/modules/rawdatamethods/.DS_Store
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-85> (0)
>    - *M* src/main/java/net/sf/mzmine/modules/rawdatamethods/
>    peakpicking/manual/ManualPeak.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-86> (15)
>    - *M* src/main/java/net/sf/mzmine/modules/rawdatamethods/
>    peakpicking/manual/ManualPickerTask.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-87> (1)
>    - *M* src/main/java/net/sf/mzmine/modules/rawdatamethods/
>    peakpicking/massdetection/MassDetectionParameters.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-88> (15)
>    - *M* src/main/java/net/sf/mzmine/modules/rawdatamethods/
>    peakpicking/massdetection/MassDetectionTask.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-89> (234)
>    - *M* src/main/java/net/sf/mzmine/modules/rawdatamethods/
>    peakpicking/massdetection/exactmass/ExactMzDataPoint.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-90> (4)
>    - *A* src/main/java/net/sf/mzmine/modules/visualization/.DS_Store
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-91> (0)
>    - *M* src/main/java/net/sf/mzmine/modules/visualization/peaklisttable/
>    PeakListTablePopupMenu.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-92> (15)
>    - *A* src/main/java/net/sf/mzmine/modules/visualization/
>    pointtwod/.DS_Store
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-93> (0)
>    - *A* src/main/java/net/sf/mzmine/modules/visualization/
>    pointtwod/PointPeakDataPoint.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-94> (57)
>    - *A* src/main/java/net/sf/mzmine/modules/visualization/pointtwod/
>    PointPeakDataRenderer.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-95> (66)
>    - *A* src/main/java/net/sf/mzmine/modules/visualization/
>    pointtwod/PointPeakDataSet.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-96> (124)
>    - *A* src/main/java/net/sf/mzmine/modules/visualization/pointtwod/
>    PointPeakToolTipGenerator.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-97> (69)
>    - *A* src/main/java/net/sf/mzmine/modules/visualization/
>    pointtwod/PointPlotMode.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-98> (24)
>    - *A* src/main/java/net/sf/mzmine/modules/visualization/pointtwod/
>    PointTwoDBottomPanel.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-99> (341)
>    - *A* src/main/java/net/sf/mzmine/modules/visualization/
>    pointtwod/PointTwoDDataSet.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-100> (384)
>    - *A* src/main/java/net/sf/mzmine/modules/visualization/pointtwod/
>    PointTwoDPaletteType.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-101> (94)
>    - *A* src/main/java/net/sf/mzmine/modules/visualization/
>    pointtwod/PointTwoDPlot.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-102> (302)
>    - *A* src/main/java/net/sf/mzmine/modules/visualization/
>    pointtwod/PointTwoDToolBar.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-103> (117)
>    - *A* src/main/java/net/sf/mzmine/modules/visualization/pointtwod/
>    PointTwoDVisualizerModule.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-104> (133)
>    - *A* src/main/java/net/sf/mzmine/modules/visualization/pointtwod/
>    PointTwoDVisualizerParameters.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-105> (55)
>    - *A* src/main/java/net/sf/mzmine/modules/visualization/pointtwod/
>    PointTwoDVisualizerWindow.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-106> (178)
>    - *A* src/main/java/net/sf/mzmine/modules/visualization/
>    pointtwod/PointTwoDXYPlot.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-107> (322)
>    - *A* src/main/java/net/sf/mzmine/modules/visualization/
>    pointtwod/help/2D_allpeaks.png
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-108> (0)
>    - *A* src/main/java/net/sf/mzmine/modules/visualization/
>    pointtwod/help/2D_axes.png
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-109> (0)
>    - *A* src/main/java/net/sf/mzmine/modules/visualization/
>    pointtwod/help/2D_centroid.png
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-110> (0)
>    - *A* src/main/java/net/sf/mzmine/modules/visualization/
>    pointtwod/help/2D_continuous.png
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-111> (0)
>    - *A* src/main/java/net/sf/mzmine/modules/visualization/
>    pointtwod/help/2D_palette_color.png
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-112> (0)
>    - *A* src/main/java/net/sf/mzmine/modules/visualization/
>    pointtwod/help/2D_peak.png
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-113> (0)
>    - *A* src/main/java/net/sf/mzmine/modules/visualization/
>    pointtwod/help/2D_peak_tooltip.png
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-114> (0)
>    - *A* src/main/java/net/sf/mzmine/modules/visualization/
>    pointtwod/help/2D_peaks_datapoints.png
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-115> (0)
>    - *A* src/main/java/net/sf/mzmine/modules/visualization/
>    pointtwod/help/2D_threshold_peaks.png
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-116> (0)
>    - *A* src/main/java/net/sf/mzmine/modules/visualization/
>    pointtwod/help/2D_toolbar.png
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-117> (0)
>    - *A* src/main/java/net/sf/mzmine/modules/visualization/
>    pointtwod/help/2D_zoom.png
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-118> (0)
>    - *A* src/main/java/net/sf/mzmine/modules/visualization/
>    pointtwod/help/help.html
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-119> (105)
>    - *A* src/main/java/net/sf/mzmine/modules/visualization/
>    pointtwod/tmpkeep/PeakDataPoint.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-120> (57)
>    - *A* src/main/java/net/sf/mzmine/modules/visualization/
>    pointtwod/tmpkeep/PeakThresholdMode.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-121> (42)
>    - *A* src/main/java/net/sf/mzmine/modules/visualization/
>    pointtwod/tmpkeep/PeakThresholdParameter.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-122> (125)
>    - *M* src/main/java/net/sf/mzmine/modules/visualization/spectra/
>    SpectraVisualizerModule.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-123> (19)
>    - *M* src/main/java/net/sf/mzmine/modules/visualization/spectra/
>    SpectraVisualizerWindow.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-124> (14)
>    - *M* src/main/java/net/sf/mzmine/modules/visualization/twod/
>    TwoDPlot.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-125> (3)
>    - *A* src/main/java/net/sf/mzmine/modules/visualization/twod/
>    help/help.html~HEAD
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-126> (105)
>    - *A* src/main/java/net/sf/mzmine/modules/visualization/twod/
>    help/help.html~HEAD_0
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-127> (105)
>    - *A* src/main/java/net/sf/mzmine/modules/visualization/twod/
>    help/help.html~treerangeset
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-128> (105)
>    - *A* src/main/java/net/sf/mzmine/modules/visualization/twod/
>    help/help.html~treerangeset_0
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-129> (105)
>    - *M* src/main/java/net/sf/mzmine/parameters/dialogs/
>    ParameterSetupDialog.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-130> (1)
>    - *A* src/main/java/net/sf/mzmine/parameters/parametertypes/.
>    LCKParameterSetComponents.java~
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-131> (1)
>    - *A* src/main/java/net/sf/mzmine/parameters/parametertypes/
>    ParameterSetComponent.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-132> (97)
>    - *A* src/main/java/net/sf/mzmine/parameters/parametertypes/
>    ParameterSetParameter.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-133> (138)
>    - *A* src/main/java/net/sf/mzmine/parameters/parametertypes/ranges/
>    ListDoubleRangeComponent.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-134> (116)
>    - *A* src/main/java/net/sf/mzmine/parameters/parametertypes/ranges/
>    ListDoubleRangeParameter.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-135> (115)
>    - *M* src/main/java/net/sf/mzmine/project/impl/RawDataFileImpl.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-136> (5)
>    - *A* src/main/java/net/sf/mzmine/util/adap/ADAPInterface.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-137> (212)
>    - *M* src/main/java/net/sf/mzmine/util/components/
>    CombinedXICComponent.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-138> (29)
>    - *M* src/main/java/net/sf/mzmine/util/components/
>    PeakSummaryComponent.java
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-139> (99)
>    - *A* src/main/lib/dulab/adap/0.0.1/adap-0.0.1.jar
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-140> (0)
>    - *A* src/main/lib/dulab/adap/0.0.1/adap-0.0.1.pom
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-141> (68)
>    - *A* src/main/lib/jmzml/jmzml-1.7.5.jar
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-142> (0)
>    - *A* src/main/lib/uk/ac/ebi/jmzml/jmzml/1.7.3/_maven.repositories
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-143> (3)
>    - *A* src/main/lib/uk/ac/ebi/jmzml/jmzml/1.7.3/jmzml-1.7.3.pom
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-144> (359)
>    - *A* src/main/lib/uk/ac/ebi/jmzml/jmzml/1.7.3/jmzml-1.7.3.pom.
>    lastUpdated <https://github.com/mzmine/mzmine2/pull/278/files#diff-145>
>    (7)
>    - *A* src/main/lib/uk/ac/ebi/jmzml/jmzml/1.7.3/jmzml-1.7.3.pom.sha1
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-146> (1)
>    - *A* src/main/lib/uk/ac/ebi/jmzml/jmzml/1.7.4/_maven.repositories
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-147> (3)
>    - *A* src/main/lib/uk/ac/ebi/jmzml/jmzml/1.7.4/_remote.repositories
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-148> (3)
>    - *A* src/main/lib/uk/ac/ebi/jmzml/jmzml/1.7.4/jmzml-1.7.4.jar
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-149> (0)
>    - *A* src/main/lib/uk/ac/ebi/jmzml/jmzml/1.7.4/jmzml-1.7.4.jar.sha1
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-150> (1)
>    - *A* src/main/lib/uk/ac/ebi/jmzml/jmzml/1.7.4/jmzml-1.7.4.pom
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-151> (364)
>    - *A* src/main/lib/uk/ac/ebi/jmzml/jmzml/1.7.4/jmzml-1.7.4.pom.
>    lastUpdated <https://github.com/mzmine/mzmine2/pull/278/files#diff-152>
>    (18)
>    - *A* src/main/lib/uk/ac/ebi/jmzml/jmzml/1.7.4/jmzml-1.7.4.pom.sha1
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-153> (1)
>    - *A* src/main/lib/uk/ac/ebi/jmzml/jmzml/1.7.5/_remote.repositories
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-154> (4)
>    - *A* src/main/lib/uk/ac/ebi/jmzml/jmzml/1.7.5/jmzml-1.7.5.jar
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-155> (0)
>    - *A* src/main/lib/uk/ac/ebi/jmzml/jmzml/1.7.5/jmzml-1.7.5.jar.sha1
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-156> (1)
>    - *A* src/main/lib/uk/ac/ebi/jmzml/jmzml/1.7.5/jmzml-1.7.5.pom
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-157> (364)
>    - *A* src/main/lib/uk/ac/ebi/jmzml/jmzml/1.7.5/jmzml-1.7.5.pom.
>    lastUpdated <https://github.com/mzmine/mzmine2/pull/278/files#diff-158>
>    (7)
>    - *A* src/main/lib/uk/ac/ebi/jmzml/jmzml/1.7.5/jmzml-1.7.5.pom.sha1
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-159> (1)
>    - *A* src/main/lib/uk/ac/ebi/pride/jmztab-modular-model/3.0.2/_
>    maven.repositories
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-160> (4)
>    - *A* src/main/lib/uk/ac/ebi/pride/jmztab-modular-model/3.0.2/
>    jmztab-modular-model-3.0.2.jar
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-161> (0)
>    - *A* src/main/lib/uk/ac/ebi/pride/jmztab-modular-model/3.0.2/
>    jmztab-modular-model-3.0.2.jar.sha1
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-162> (1)
>    - *A* src/main/lib/uk/ac/ebi/pride/jmztab-modular-model/3.0.2/
>    jmztab-modular-model-3.0.2.pom
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-163> (51)
>    - *A* src/main/lib/uk/ac/ebi/pride/jmztab-modular-model/3.0.2/
>    jmztab-modular-model-3.0.2.pom.lastUpdated
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-164> (12)
>    - *A* src/main/lib/uk/ac/ebi/pride/jmztab-modular-model/3.0.2/
>    jmztab-modular-model-3.0.2.pom.sha1
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-165> (1)
>    - *A* src/main/lib/uk/ac/ebi/pride/jmztab-modular-model/3.0.3/_
>    remote.repositories
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-166> (4)
>    - *A* src/main/lib/uk/ac/ebi/pride/jmztab-modular-model/3.0.3/
>    jmztab-modular-model-3.0.3.jar
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-167> (0)
>    - *A* src/main/lib/uk/ac/ebi/pride/jmztab-modular-model/3.0.3/
>    jmztab-modular-model-3.0.3.jar.sha1
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-168> (1)
>    - *A* src/main/lib/uk/ac/ebi/pride/jmztab-modular-model/3.0.3/
>    jmztab-modular-model-3.0.3.pom
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-169> (51)
>    - *A* src/main/lib/uk/ac/ebi/pride/jmztab-modular-model/3.0.3/
>    jmztab-modular-model-3.0.3.pom.lastUpdated
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-170> (7)
>    - *A* src/main/lib/uk/ac/ebi/pride/jmztab-modular-model/3.0.3/
>    jmztab-modular-model-3.0.3.pom.sha1
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-171> (1)
>    - *A* src/main/lib/uk/ac/ebi/pride/jmztab-modular-util/3.0.2/_
>    maven.repositories
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-172> (4)
>    - *A* src/main/lib/uk/ac/ebi/pride/jmztab-modular-util/3.0.2/
>    jmztab-modular-util-3.0.2.jar
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-173> (0)
>    - *A* src/main/lib/uk/ac/ebi/pride/jmztab-modular-util/3.0.2/
>    jmztab-modular-util-3.0.2.jar.sha1
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-174> (1)
>    - *A* src/main/lib/uk/ac/ebi/pride/jmztab-modular-util/3.0.2/
>    jmztab-modular-util-3.0.2.pom
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-175> (57)
>    - *A* src/main/lib/uk/ac/ebi/pride/jmztab-modular-util/3.0.2/
>    jmztab-modular-util-3.0.2.pom.lastUpdated
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-176> (12)
>    - *A* src/main/lib/uk/ac/ebi/pride/jmztab-modular-util/3.0.2/
>    jmztab-modular-util-3.0.2.pom.sha1
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-177> (1)
>    - *A* src/main/lib/uk/ac/ebi/pride/jmztab-modular-util/3.0.3/_
>    remote.repositories
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-178> (4)
>    - *A* src/main/lib/uk/ac/ebi/pride/jmztab-modular-util/3.0.3/
>    jmztab-modular-util-3.0.3.jar
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-179> (0)
>    - *A* src/main/lib/uk/ac/ebi/pride/jmztab-modular-util/3.0.3/
>    jmztab-modular-util-3.0.3.jar.sha1
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-180> (1)
>    - *A* src/main/lib/uk/ac/ebi/pride/jmztab-modular-util/3.0.3/
>    jmztab-modular-util-3.0.3.pom
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-181> (61)
>    - *A* src/main/lib/uk/ac/ebi/pride/jmztab-modular-util/3.0.3/
>    jmztab-modular-util-3.0.3.pom.lastUpdated
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-182> (7)
>    - *A* src/main/lib/uk/ac/ebi/pride/jmztab-modular-util/3.0.3/
>    jmztab-modular-util-3.0.3.pom.sha1
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-183> (1)
>    - *A* src/main/lib/uk/ac/ebi/pride/jmztab-modular/3.0.2/_maven.
>    repositories
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-184> (3)
>    - *A* src/main/lib/uk/ac/ebi/pride/jmztab-modular/3.0.2/jmztab-
>    modular-3.0.2.pom
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-185> (348)
>    - *A* src/main/lib/uk/ac/ebi/pride/jmztab-modular/3.0.2/jmztab-
>    modular-3.0.2.pom.sha1
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-186> (1)
>    - *A* src/main/lib/uk/ac/ebi/pride/jmztab-modular/3.0.3/_remote.
>    repositories
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-187> (3)
>    - *A* src/main/lib/uk/ac/ebi/pride/jmztab-modular/3.0.3/jmztab-
>    modular-3.0.3.pom
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-188> (355)
>    - *A* src/main/lib/uk/ac/ebi/pride/jmztab-modular/3.0.3/jmztab-
>    modular-3.0.3.pom.sha1
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-189> (1)
>    - *M* src/main/lib/vendor_lib/waters/WatersRawDump.cpp
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-190> (110)
>    - *M* src/main/lib/vendor_lib/waters/include/
>    MassLynxRawChromatogramReader.h
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-191> (246)
>    - *M* src/main/lib/vendor_lib/waters/include/MassLynxRawDataFile.h
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-192> (118)
>    - *M* src/main/lib/vendor_lib/waters/include/MassLynxRawDefs.h
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-193> (15)
>    - *M* src/main/lib/vendor_lib/waters/include/MassLynxRawInfo.h
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-194> (221)
>    - *M* src/main/lib/vendor_lib/waters/include/MassLynxRawReader.h
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-195> (127)
>    - *M* src/main/lib/vendor_lib/waters/include/MassLynxRawScanReader.h
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-196> (86)
>    - *M* src/main/lib/vendor_lib/waters/include/
>    MassLynxRawScanStatsReader.h
>    <https://github.com/mzmine/mzmine2/pull/278/files#diff-197> (119)
>
> Patch Links:
>
>    - https://github.com/mzmine/mzmine2/pull/278.patch
>    - https://github.com/mzmine/mzmine2/pull/278.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mzmine/mzmine2/pull/278>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABtqVOZdfa9QVmrqIPmHcpvQ6eBYU71zks5rjv9lgaJpZM4MXMZv>
> .
>

du-lab commented:
Hi Tomas,

I apologize for my late reply to your questions you raised 11 days ago. I was busy with a proposal and just got time to answer your questions. 

Your questions have made us aware of some changes that we should have made. We have incorporated the changes (described below) and they are now a part of our master branch. Let us know if there is anything else we can do and if we should make another pull request to incorporate these changes.

1) We added some packages temporarily when we were having some problems accessing the maven online repositories and forgot to remove them. They have been removed but we have left the dulab lib.

2) Some of the source code is included in the source tree. The heart of the ADAP algorithms are part of separate ADAP package which resides in the jar file. We prefer to develop ADAP as a separate package because we might want to use it separately. The new code in the MZmine 2 source tree is the interface to the ADAP package except the ADAP chromatogram building, some additional classes, and visualization which are directly implemented in MZmine 2.

3) The commented out modules in MZmineModulesList.java are currently working but have not been fully tested. We have just removed these and will incorporate them into MZmine 2 some time later.

I will look into your change request and get back to you.

Thank you very much for taking time reviewing our pull request!

Xiuxia

du-lab commented:
Hi Tomas,

My group has made modifications you specified and pushed to MZmine 2 that I forked.

Please let me know if I need to make a new pull request or not.

Thank you and the MZmine 2 team for reviewing the changes.

Xiuxia

tomas-pluskal commented:
Hi Xiuxia,
Thanks! The pull request looks much cleaner now.
One last thing: the PR seems to include some changes to the waters raw file reader (src/main/lib/vendor_lib/waters/WatersRawDump.cpp etc). The changes actually corrupt the files, as you can see in the diff. Could you please revert these corrupted files to the original ones?

tomas-pluskal commented:
Hi, although the WatersRawDump.cpp file was fixed, the PR still contains corrupting changes to many other files in src/main/lib/vendor_lib/waters/. Could you please fix those, too?

tomas-pluskal commented:
Hi, 
I noticed there are some more bogus files in the PR:
````
src/main/java/net/sf/mzmine/modules/visualization/twod/help/help.html~treerangeset
src/main/java/net/sf/mzmine/modules/visualization/twod/help/help.html~treerangeset_0
src/main/java/net/sf/mzmine/modules/masslistmethods/ADAPchromatogrambuilder/.LCKChromatogramBuilderParameters.java~
````

tomas-pluskal commented:
Thanks for cleaning up. One more question: why is Ms2SearchModule removed from the module list?

tomas-pluskal commented:
One more bogus file: src/main/java/net/sf/mzmine/modules/masslistmethods/.DS_Store

tomas-pluskal commented:
there are several .DS_Store files, actually: src/main/java/net/sf/mzmine/modules/rawdatamethods/.DS_Store

tomas-pluskal commented:
Thanks, I'll merge this PR now!
I will test it and please do the same. 
Perhaps I can release a new MZmine version in a few days.

tomas-pluskal commented:
So, after merging the code, there are 42 compilation errors. Could you please take a look at the current master branch and submit another PR with fixes?

du-lab commented:
Hi Tomas,

Using JDK 1.8 and maven3 we have successfully compiled the merged MZmine 2 master (Windows 7E, Linux (debian), and OSX 10.11.6). Using JDK 1.7 and maven3 we got 4 errors regarding the getOrDefault method. I'm wondering if you could please send us the details of how you are compiling so we could recreate the errors you are getting. Also maybe send the full output in case we have trouble recreating the problem?

Xiuxia on behalf of the Du-lab Team

tomas-pluskal commented:
Hi Xiuxia,

I am using Eclipse to compile the project, but I suppose the errors that
Eclipse is reporting are caused by the same error that you are seeing
(getOrDefault method), because the target platform is set to version 1.7.
If you could avoid using Java 8 methods for the time being, that would be
great.

Best,

Tomas



On Wed, Mar 29, 2017 at 1:40 PM, du-lab <notifications@github.com> wrote:

> Hi Tomas,
>
> Using JDK 1.8 and maven3 we have successfully compiled the merged MZmine 2
> master (Windows 7E, Linux (debian), and OSX 10.11.6). Using JDK 1.7 and
> maven3 we got 4 errors regarding the getOrDefault method. I'm wondering if
> you could please send us the details of how you are compiling so we could
> recreate the errors you are getting. Also maybe send the full output in
> case we have trouble recreating the problem?
>
> Xiuxia on behalf of the Du-lab Team
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/mzmine/mzmine2/pull/278#issuecomment-290166260>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ABtqVE2oJ_yVO_Gmykxj8FOdJI7ddXBNks5rqpeDgaJpZM4MXMZv>
> .
>
